app/src/main/java/com/example/grocerieswizard/shop/ShopHelperImpl.java

```diff
@@ -1,51 +1,19 @@
-package com.example.grocerieswizard.activities;
+package com.example.grocerieswizard.shop;
 
-import android.os.Bundle;
-import android.util.Log;
-import android.view.View;
+import static androidx.constraintlayout.helper.widget.MotionEffect.TAG;
 
-import androidx.appcompat.app.AppCompatActivity;
-import androidx.recyclerview.widget.LinearLayoutManager;
+import android.util.Log;
 
-import com.example.grocerieswizard.RecipeDatabaseHelper;
-import com.example.grocerieswizard.adapters.ShopAdapter;
-import com.example.grocerieswizard.databinding.ActivityShoppingMenuBinding;
-import com.example.grocerieswizard.interfaces.ShopInterface;
 import com.example.grocerieswizard.models.IngredientModel;
 import com.example.grocerieswizard.models.RecipeModel;
-import com.example.grocerieswizard.models.ShoppingItem;
-import com.example.grocerieswizard.models.SubShoppingItem;
+import com.example.grocerieswizard.shop.subshop.SubShoppingItem;
 
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Objects;
 
-public class ShoppingMenuActivity extends AppCompatActivity implements ShopInterface {
-
-    private static final String TAG = "ShoppingMenuActivity";
-    ActivityShoppingMenuBinding binding;
-
-    @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        binding = ActivityShoppingMenuBinding.inflate(getLayoutInflater());
-        View view = binding.getRoot();
-        setContentView(view);
-
-        ArrayList<RecipeModel> recipes;
-        ShopAdapter adapter;
-        try (RecipeDatabaseHelper dbHelper = new RecipeDatabaseHelper(this)) {
-
-            adapter = new ShopAdapter(this);
-            binding.shoppingCart.setAdapter(adapter);
-            binding.shoppingCart.setLayoutManager(new LinearLayoutManager(this));
-
-            adapter.setShopInterface(this);
-            recipes = dbHelper.getSelectedRecipes();
-        }
-        adapter.setSelectedRecipeList(recipes);
-    }
+public class ShopHelperImpl implements ShopHelper {
```

Comment:
Having an interface for doing some work is great, but don't implement it in Activity or Fragment or Adapter if the calculations are not dependent. Good practice is to just have implementation of the interface.